core/src/main/scala/org/apache/spark/scheduler/ExecutorDecommissionInfo.scala

```diff
@@ -25,7 +25,19 @@ package org.apache.spark.scheduler
  *                shuffle data might be lost even if the external shuffle service is enabled.
  */
 private[spark]
-case class ExecutorDecommissionInfo(message: String, workerHost: Option[String] = None)
+abstract class DecommissionInfo
+
+private[spark]
+case class ExecutorDecommissionInfo(
```

Comment:
hmm... Who triggered this? And I think we should  add some comments for these two new classes

Reply:
`Executor` will triggered this. Maybe we can see this comment https://github.com/apache/spark/blob/1c0bd1f9f813a341bbfdecb2c5ccde7fbc1bac2d/core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala#L109

Reply:
Could we also rename this one to `TriggeredByXXXDecommissionInfo`?

Reply:
Yes, how about `ExecutorTriggerExecutorDecommissionInfo` and `DriverTriggerExecutorDecommissionInfo`? Or could you give me a better opinion ?

Reply:
fine to me, it's good to be able to distinguish them clearly